tubular/imputers.py

```diff
@@ -517,18 +522,19 @@ def __init__(
     ) -> None:
         super().__init__(columns=columns, **kwargs)
 
-    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+    @nw.narwhalify
+    def transform(self, X: FrameT) -> FrameT:
         """Create new columns indicating the position of null values for each variable in self.columns.
 
         Parameters
         ----------
-        X : pd.DataFrame
+        X : FrameT
             Data to add indicators to.
 
         """
-        X = super().transform(X)
+        X = nw.from_native(super().transform(X))
 
         for c in self.columns:
-            X[f"{c}_nulls"] = X[c].isna().astype(np.int8)
+            X = X.with_columns((X[c].is_null()).cast(nw.Int8).alias(f"{c}_nulls"))
```

Comment:
Q: think polars supports the 2bit Boolean type (unlike pandas which I think stores bools with 8bit memory) - curious what happens if you use nw.Boolean instead of nw.Int8 here?

Reply:
(might be out of scope for this PR, but interested to inform a future issue)

Reply:
apparently should try and avoid using pandas-like syntax for speed reasons, so would switch `X[c].is_null()...` with `nw.col(c).is_null()...`

Reply:
Ahh okay cool [here](https://github.com/lvgig/tubular/pull/337/commits/ddbf0bf7a9519f4b557680703ce606baeb9613ba)